ocs_ci/deployment/deployment.py

```diff
@@ -678,3 +696,149 @@ def create_catalog_source(image=None, ignore_upgrade=False):
+    ))
+    # Update local-storage-operator subscription data with channel
+    for data in lso_data:
+        if data['kind'] == 'Subscription':
+            data['spec']['channel'] = get_ocp_version()
+
+    # Create temp yaml file and create local storage operator
+    logger.info(
+        "Creating temp yaml file with local-storage-operator data:\n %s",
+        lso_data
+    )
+    lso_data_yaml = tempfile.NamedTemporaryFile(
+        mode='w+', prefix='local_storage_operator', delete=False
+    )
+    templating.dump_data_to_temp_yaml(
+        lso_data, lso_data_yaml.name
+    )
+    logger.info("Creating local-storage-operator")
+    run_cmd(f"oc create -f {lso_data_yaml.name}")
+
+    local_storage_operator = ocp.OCP(
+        kind=constants.POD, namespace='local-storage'
+    )
+    assert local_storage_operator.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=constants.LOCAL_STORAGE_OPERATOR_LABEL,
+        timeout=600
+    ), "Local storage operator did not reach running phase"
+
+    # Retrieve NVME device path ID for each worker node
+    device_paths = get_device_paths(worker_names)
+
+    # Pull local volume yaml data
+    logger.info("Pulling LocalVolume CR data from yaml")
+    lv_data = templating.load_yaml(
+        constants.LOCAL_VOLUME_YAML
+    )
+
+    # Update local volume data with NVME IDs
+    logger.info(
+        "Updating LocalVolume CR data with device paths: %s",
+        device_paths
+    )
+    lv_data['spec']['storageClassDevices'][0][
+        'devicePaths'
+    ] = device_paths
+
+    # Create temp yaml file and create local volume
+    lv_data_yaml = tempfile.NamedTemporaryFile(
+        mode='w+', prefix='local_volume', delete=False
+    )
+    templating.dump_data_to_temp_yaml(
+        lv_data, lv_data_yaml.name
+    )
+    logger.info("Creating LocalVolume CR")
+    run_cmd(f"oc create -f {lv_data_yaml.name}")
+    logger.info("Waiting 30 seconds for PVs to create")
+    verify_pvs_created(len(worker_names))
+
+
+@retry(AssertionError, 12, 10, 1)
+def verify_pvs_created(num_workers):
+    """
+    Verify that PVs were created and are in the Available state
+
+    Args:
+        num_workers (int): number of worker nodes in the cluster
+
+    Raises:
+        AssertionError: if any PVs are not in the Available state or if the
+            number of PVs does not match the number of workers in the cluster.
+
+    """
+    logger.info("Verifying PVs are created")
+    out = run_cmd("oc get pv -o json")
+    pv_json = json.loads(out)
+    for pv in pv_json['items']:
+        pv_state = pv['status']['phase']
+        pv_name = pv['metadata']['name']
+        logger.info("%s is %s", pv_name, pv_state)
+        assert pv_state == 'Available', (
+            f"{pv_name} not in 'Available' state. Current state is {pv_state}"
+        )
+    num_pvs = len(pv_json['items'])
+    logger.debug("PVs, Workers: %s, %s", num_pvs, num_workers)
+
+
+def get_device_paths(worker_names):
```

Comment:
@clacroix12 are you planning on creating classes for the different LSO types, like we have for AWSIPI, VSPHEREUPI, etc. (maybe AWSLSO, VSPHERELSO, BMLSO?). 
And, as raised by Raz in https://github.com/red-hat-storage/ocs-ci/pull/1704/files#r397939609, having this function being implemented in the corresponding deployment class

Reply:
I think it will be overkill. If the only thing that changes is how we retrieve the device path we don't really need separate classes to handle it. This function can simply have conditionals. I think it depends on how much changes between the platform types.

If the logic changes enough to warrant breaking it out into separate class methods then that can absolutely be done but right now I don't have enough to go off of to make that determination.

Reply:
Yes, sounds good. For now, conditions in this function will do the job

Reply:
@clacroix12 We will need one more change for VMware.
For Mon either we have to create and attach a 10GB VMDK and create the Filesystem LocalVolume CR 
Or use default thin storageclass for automatic creation and attaching VMDKs